codes/block_definitions/utilities/operators_TensorFlow_Keras.py

```diff
@@ -11,31 +11,63 @@
 '''
 
 
-### packages
+# packages
+from codes.utilities.custom_logging import ezLogging
+from codes.block_definitions.utilities import argument_types
 import tensorflow as tf
 import numpy as np
 
-### sys relative to root dir
+# sys relative to root dir
 import sys
 from os.path import dirname, realpath
 sys.path.append(dirname(dirname(dirname(dirname(realpath(__file__))))))
 
-### abosulte imports wrt root
-from codes.block_definitions.utilities import argument_types
```

Comment:
I would keep these 'imports' here after we add the current directory to the sys.path

almost every single one of our py scripts are like this: import external modules, add root dir to sys.path, import local py scripts relative to root dir.

technically once one py script adds root dir to the sys.path then none of the others need to and then it doesn't matter where (ie could be before or could be after we alter sys.path) we import the local scripts, but we keep it like this for consistency